### PR TITLE
Revert doc-comments change to the default profile

### PR DESCRIPTION
**Pull request by Julow**
This surprised users.

We should release 0.14.1 with this patch.

cc @samoht

### CODE REVIEW

**Comment by samoht** [comment]
Why does it change the formatting of vendored code?

appart that LGTM!

File: CHANGES.md [hunk 1]
```diff
@@ -1,3 +1,10 @@
+### unreleased
+
+#### Changes
+
+  + The default for `doc-comments` is changed to `after` (#1335) (Jules Aguillon)
+    This revert a change introduced in 0.14.0.
```
Comment (gpetiot) [inline CHANGES.md:6]: Can you mention #1325 here so we can easily jump back and forth between the PRs ? thanks

**Comment by gpetiot** [comment]
> Why does it change the formatting of vendored code?

That's because the vendored code uses the `conventional` profile (as I made it a separate library before), as opposed to the rest of ocamlformat code using the `ocamlformat` profile (that has not been rolledback)

File: CHANGES.md (hunk 1 above)
Comment (Julow) [reply CHANGES.md]: The change is from https://github.com/ocaml-ppx/ocamlformat/pull/1012. The change to the `ocamlformat` profile is not reverted by this PR.

**Comment by Julow** [comment]
We should probably disable formatting of vendored code. This makes using `git subtree` impossible and unecessary diffs.

**Comment by gpetiot** [comment]
> We should probably disable formatting of vendored code. This makes using `git subtree` impossible and unecessary diffs.

I was not aware of that, sure we can disable it.

**Pull request merged**